omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,220 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_8;
+
+import org.openmrs.api.APIException;
+import org.openmrs.module.webservices.helper.TaskAction;
+import org.openmrs.module.webservices.helper.TaskServiceWrapper;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.ConversionUtil;
+import org.openmrs.module.webservices.rest.web.RequestContext;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@Resource(name = RestConstants.VERSION_1 + "/taskaction", supportedClass = TaskAction.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class TaskActionResource1_8 extends BaseDelegatingResource<TaskAction> implements Creatable {
+	
+	private TaskServiceWrapper taskServiceWrapper = new TaskServiceWrapper();
+	
+	public void setTaskServiceWrapper(TaskServiceWrapper taskServiceWrapper) {
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks() != null && action.isAllTasks()) {
```

Comment:
Why do you test for action.isAllTasks() != null?

Reply:
here for some actions, I will include `{ "allTasks"  : true }` in the REST body. So here I am checking that parameter for the confirmation. (Common actions only need this parameter, others can skip this)

Reply:
Should isAllTasks() return null? I though it is a boolean function returning either true or false.

Reply:
If the user's doesn't mention in the REST request, then It may be null. 
Like for this request, 

`{ "action" : "shutdowntask", "tasks" : ["taskA"] }`

Reply:
By the way, why would the client have to mention allTasks? Why would i need to pass a list of tasks if am not interested in all of them?